src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
+
+    this.logger.debug(
+      `Request Method:[${req.method}] || Request URI:${
+        req.baseUrl
+      } || Request Body:[${JSON.stringify(
+        req.body,
+      )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
+        req.ip
+      }] || User Details: [${JSON.stringify(userDetail)}]`,
+    );
```

Comment:
PullProofAI

The `LoggerMiddleware` class implementation looks good. It captures request details and user information for logging purposes. However, there are a few suggestions to enhance the code:

1. **Security Enhancement**: Ensure that sensitive user information is not logged unintentionally. Consider filtering out or masking certain fields like passwords or tokens before logging.

2. **Error Handling**: Add error handling to gracefully handle any exceptions that might occur during decoding the JWT token or logging process.

3. **Consistency**: Maintain consistency in formatting the log messages for better readability and debugging.

4. **Modularity**: Consider extracting the logic for decoding the JWT token and logging into separate methods for better modularity and testability.

Overall, the middleware provides valuable request logging functionality but can be improved further for security and maintainability.

```diff
+ // Example of filtering sensitive information before logging
+ const filteredUserDetail = { ...userDetail };
+ delete filteredUserDetail.password;
+
+ // Add error handling for JWT decoding and logging
+ try {
+     // Decode JWT token
+     const decodedToken = this.jwt.decode(token);
+     // Log user details
+     this.logUserDetails(req, userDetail);
+ } catch (error) {
+     this.logger.error(`Error occurred while decoding token: ${error.message}`);
+ }
+
+ // Extract logging logic into a separate method
+ private logUserDetails(req: Request, userDetail: any) {
+     this.logger.debug(
+         `Request Method:[${req.method}] || Request URI:${req.baseUrl} || Request Body:[${JSON.stringify(req.body)}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${req.ip}] || User Details: [${JSON.stringify(userDetail)}]`,
+     );
+ }
```



<!-- This is an auto-generated comment by PullProofAI -->